pkg/controllers/cloudshell_controller.go

```diff
@@ -449,14 +449,16 @@ func (c *Controller) CreateRouteRule(ctx context.Context, cloudshell *cloudshell
 // GetMasterNodeIP could find the one master node IP.
 func (c *Controller) GetMasterNodeIP(ctx context.Context) (string, error) {
 	// the label "node-role.kubernetes.io/master" be removed in k8s 1.24, and replace with
-	// "node-role.kubernetes.io/cotrol-plane".
+	// "node-role.kubernetes.io/control-plane".
 	nodes := &corev1.NodeList{}
 	if err := c.List(ctx, nodes, client.MatchingLabels{"node-role.kubernetes.io/master": ""}); err != nil {
 		return "", err
 	}
 	if len(nodes.Items) == 0 {
 		if err := c.List(ctx, nodes, client.MatchingLabels{"node-role.kubernetes.io/control-plane": ""}); err != nil || len(nodes.Items) == 0 {
-			return "", err
+			if err != nil {
```

Comment:
Why do we need to determine if err is empty again?

Reply:
> Why do we need to determine if err is empty again?

I mistakenly missed the inline checking, the updated is remove unnecessary length checking.

Reply:
Logically, this code isn't necessary. But I think keeping this code might make the meaning clearer.

Reply:
I think it's not necessary, maybe it can be closed.

Reply:
other fixes are needed. You can update it if you feel like continuing.